### ISSUE TITLE
Accessibility: Select onMouseDown event outside of Select Menu fails WCAG 2.5.2 Pointer Cancellation

### ISSUE THREAD

**Issue by Watso196** [radix-ui/primitives#3146]
### Current Behavior

When the Select menu is open, and the user clicks outside of the Select menu content to close it, this event fires onPointerDown. This means that as soon as the mouse button is depressed, the Select menu closes.

### Expected behavior

When the Select menu is open, and the user clicks outside of the Select menu content to close it, this event should fire onPointerUp. This means that the Select menu will not close until the user releases their mouse button outside of the Select content. This allows users to cancel the mouse click operation, and meets [WCAG 2.5.2 Pointer Cancellation](https://www.w3.org/WAI/WCAG21/Understanding/pointer-cancellation.html). 

### Reproducible example

[CodeSandbox Example](https://codesandbox.io/p/sandbox/gracious-star-t4nkyj?workspaceId=21935fec-e5e3-4a93-92b6-4bb119a700c4)

### Suggested solution

Swap from using onPointerDown event to onPointerUp event

### Additional context

[WCAG 2.5.2 Pointer Cancellation](https://www.w3.org/WAI/WCAG21/Understanding/pointer-cancellation.html)

### Your environment

| Software         | Name(s) | Version |
| ---------------- | ------- | ------- |
| Radix Package(s) |    radix-ui-primitive, radix-ui/react-select     |     v1.0.1, v2.1.1    |
| React            | n/a     |    18.2.0     |
| Browser          |    Chrome, Firefox     |    129.0.6668.59 , 130.0.1     |
| Assistive tech   |     n/a    |         |
| Node             | n/a     |         |
| npm/yarn         |     yarn    |    1.22.17     |
| Operating System |    MacOS     |    Sonoma 14.7     |

**Comment by bolonio** [comment]
Similar issue: https://github.com/radix-ui/primitives/issues/3124
